OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldassignability/L2FieldAssignabilityAnalysis.scala

```diff
@@ -152,24 +151,45 @@ class L2FieldAssignabilityAnalysis private[analyses] (val project: SomeProject)
         if (writesInMethod.distinctBy(_._2).size > 1)
             return true; // Field is written in multiple locations, thus must be assignable
 
-        // If we have no information about the receiver, we soundly return
-        if (receiverVar.isEmpty)
+        // If we have no information about the receiver, we soundly return true
+        // However, a static field has no receiver
+        if (receiverVar.isEmpty && !state.field.isStatic)
             return true;
 
-        val assignedValueObject = receiverVar.get
-        if (assignedValueObject.definedBy.exists(_ < 0))
+        val assignedValueObject =
+            if (index > 0 && stmts(index).isPutStatic) {
+                stmts(index).asPutStatic.value.asVar
+            } else
+                receiverVar.get
+
+        // When there are more than 1 definitionsite, we soundly return true
+        if (assignedValueObject.definedBy.size != 1)
+            return true;
+
+        val definitionSite = assignedValueObject.definedBy.head
+
+        if (definitionSite < -1 ||
+            (definitionSite == -1 && !definedMethod.definedMethod.isConstructor)
+        )
             return true;
 
-        val assignedValueObjectVar = stmts(assignedValueObject.definedBy.head).asAssignment.targetVar.asVar
+        val uses = if (definitionSite == -1)
+            taCode.params.thisParameter.useSites
+        else {
+            val assignedValueObjectVar = stmts(definitionSite).asAssignment.targetVar.asVar
+            if (assignedValueObjectVar != null)
+                assignedValueObjectVar.usedBy
+            else IntTrieSet.empty
+        }
 
         val fieldWriteInMethodIndex = taCode.pcToIndex(writesInMethod.head._2)
-        if (assignedValueObjectVar != null && !assignedValueObjectVar.usedBy.forall { index =>
+        if (!uses.forall { index =>
                 val stmt = stmts(index)
 
                 fieldWriteInMethodIndex == index || // The value is itself written to another object
                     // IMPROVE: Can we use field access information to care about reflective accesses here?
                     stmt.isPutField && stmt.asPutField.name != state.field.name ||
-                    stmt.isAssignment && stmt.asAssignment.targetVar == assignedValueObjectVar ||
+                    //  stmt.isAssignment && stmt.asAssignment.targetVar == assignedValueObjectVar ||
```

Comment:
Why is this code commented out? Was this done for debugging reasons and should be reinstantiated? Or is the commented out code wrong and not needed and should be removed?